okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -281,7 +281,8 @@ private boolean shouldRetry(IOException exception, Optional<Duration> backoff) {
                 if (exception instanceof SocketTimeoutException) {
                     // non-connect timeouts should not be retried
                     SocketTimeoutException socketTimeout = (SocketTimeoutException) exception;
-                    if (!socketTimeout.getMessage().contains("connect timed out")) {
+                    if (socketTimeout.getMessage() == null
+                            || !socketTimeout.getMessage().contains("connect timed out")) {
```

Comment:
Any way we can reproduce this in a test?

Reply:
I don't have a real repro (not entirely sure how it happened, just have a trace to prove it did). I guess we could make a unit test, but don't think that'll help that much in making sure this type of regression happens again.